src/layouts/index.js

```diff
@@ -60,21 +61,24 @@ const Layout = ({ children, location: { pathname, search } }) => {
   const isAuthLayout = getLayout() === 'auth'
 
   const BootstrappedLayout = () => {
-    // show loader when user in check authorization process, not authorized yet and not on login pages
-    if (isUserLoading && !isUserAuthorized && !isAuthLayout) {
-      return null
-    }
-    if (!isUserAuthorized && /^\/admin(?=\/|$)/i.test(pathname)) {
-      return <Redirect to="/auth/admin" />
-    }
-    if (
-      !RegExp('register').test(pathname) &&
-      !isUserAuthorized &&
-      /^\/sensei(?=\/|$)/i.test(pathname)
-    ) {
-      return <Redirect to="/auth/login" />
+    if (isUserAuthorized) {
+      if (
+        (/^\/admin(?=\/|$)/i.test(pathname) && user.userType !== USER_TYPE_ENUM.ADMIN) ||
+        (/^\/sensei(?=\/|$)/i.test(pathname) && user.userType !== USER_TYPE_ENUM.SENSEI)
+      )
+        return <Redirect to="/auth/login" />
+    } else {
+      if (isUserLoading && !isUserAuthorized && !isAuthLayout) {
+        // show loader when user in check authorization process, not authorized yet and not on login pages
+        return null
```

Comment:
this condition will never be fulfilled cos it's inside the if block where `isUserAuthorised` is not falsey
```suggestion
      if (isUserAuthorized) { ... } 
      
      // at this point isUserAuthorized is falsey
      if (isUserLoading && !isAuthLayout) {
        // show loader when user in check authorization process, not authorized yet and not on login pages
        return null
```

Reply:
refer to teams for alternative to this, its not very ideal also, but no time le, i think need to leave it for now